setup.py

```diff
@@ -96,4 +96,32 @@
         ],
     },
     install_requires=requirements,
```

Comment:
@konstntokas our requirements are currently empty. This is because there where issues when installing using conda, because setup used pip to resolve dependecies after conda installed them. For some packages, pip found the PyPI version more appropriate and installed them on top of the conda version. Which broke our env in many cases. We should give it a try and see whether those issues still exist.

Reply:
FYI @pont-us  @TejasMorbagal

Reply:
Yes, worth a try!

Reply:
@forman and @TejasMorbagal: 

I used [pipreqs](https://pypi.org/project/pipreqs/) to determine the requirements. I deleted a few.

I then copied the requirements from `environment.yml` to `setup.py` for pip dependencies. I created a new conda environment with `conda env  create -f environment.yml` and installed xcube with `pip install .`. It gets problems with matplotlib-base. It cannot find a matching distribution for matplotlib-base. So I changed it to matplotlib which works fine. Any reason why we want to use matplotlib-base?

Reply:
@konstntokas Don't assume a 1:1 correspondence between PyPI and conda packages, or between either of them and the actual Python package name. In maybe >90% of cases they'll all be the same, but there are plenty of exceptions. Regarding your edits to `environment.yml`,

- `affine` is used directly in `xcube.core.geom`, so we need to keep it.
- We don't directly use `openssl`, so we can drop that. (Also, Python itself depends on openssl, so it will always be there as a transitive dependency anyway.)
- The conda package `matplotlib` includes a dependency on the Qt libraries, which are large and not needed by xcube. `matplotlib-base` excludes them, so we should stick with that. See [here](https://conda-forge.org/docs/maintainer/knowledge_base/#matplotlib).

Reply:
Thank you @forman and @pont-us 
- I added again `affine`
- I swiched back to `matplotlib-base` in `environment.yml` and kept `matplotlib` in `setup.py`, since `matplotlib-base` does not exists for `pip`. When installing with `pip` after the conda env is created from `environment.yml`, `matplotlib` is not reinstalled but `matplotlib-base` is recognized as 'Requirement already satisfied'.  So I think this is what we want, even though it is odd to me how the recognition works.

Reply:
@konstntokas Yes, it's maybe a little unexpected but pip checks for already installed dependencies directly in the Python path rather than in its own package database. But it's actually a useful behaviour when installing on top of a conda environment or a system Python with non-pip-installed packages. Nevertheless I tend to add the `--no-deps` flag, just in case, if I know that all dependencies are already present. A potential missing package is easier to deal with than a twice-installed package :)